src/App.jsx

```diff
@@ -14,45 +15,82 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function getSortedGoods(goods, sortByName, sortByLength, reverse) {
+  const preparedGoods = [...goods];
+
+  if (sortByName) {
+    preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+  }
+
+  if (sortByLength) {
+    preparedGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  return reverse ? preparedGoods.reverse() : preparedGoods;
+}
+
+export const App = () => {
+  const [sortByName, setSortByName] = useState(false);
+  const [sortByLength, setSortByLength] = useState(false);
+  const [reverse, setReverse] = useState(false);
+
+  const visibleGoods = getSortedGoods(
+    goodsFromServer, sortByName, sortByLength, reverse,
+  );
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${!sortByName ? 'is-light' : ''}`}
+          onClick={() => {
+            setSortByName(true);
```

Comment:
or even better to move this 'name' to a constant
```suggestion
            setSortBy('name');
```